apps/api/src/infrastructure/CloudFrontDistributionsManager.test.ts

```diff
@@ -0,0 +1,368 @@
+        Distribution: distributionUpdated,
+        ETag: 'test-etag',
+        $metadata: {},
+      };
+
+      mock.on(GetDistributionCommand).resolves(distributionOutputUpdated);
+      mock.on(UpdateDistributionCommand).resolves({});
+
+      await cloudFrontDistributionsManager.deleteOrigin({
+        eventId,
+      });
+
+      const getDistributionCall = mock
+        .calls()
+        .find((call) => call.args[0] instanceof GetDistributionCommand);
+      expect(getDistributionCall).toBeDefined();
+      expect(getDistributionCall?.args[0].input).toEqual({
+        Id: process.env.DISTRIBUTION_ID,
+      });
+
+      const updateCall = mock
+        .calls()
+        .find((call) => call.args[0] instanceof UpdateDistributionCommand);
+      expect(updateCall).toBeDefined();
+      const updateInput = updateCall?.args[0]
+        .input as UpdateDistributionCommandInput;
+      expect(updateInput.Id).toBe(process.env.DISTRIBUTION_ID);
+      expect(updateInput.IfMatch).toBe('test-etag');
+
+      const config = updateInput.DistributionConfig;
+
+      const origin = config?.Origins?.Items?.find((o) => o.Id === eventId);
+      expect(origin).toBeUndefined();
+      expect(config?.Origins?.Items?.length).toBe(1);
+      expect(config?.Origins?.Quantity).toBe(1);
+
+      const cacheBehaviors = config?.CacheBehaviors?.Items?.filter(
+        (b) => b.TargetOriginId === eventId
+      );
+      expect(cacheBehaviors).toHaveLength(0);
+      expect(config?.CacheBehaviors?.Items?.length).toBe(0);
+      expect(config?.CacheBehaviors?.Quantity).toBe(0);
+    });
+
+    it('should not fail if origin does not exist', async () => {
+      const { cloudFrontDistributionsManager } = setup();
+      const eventId = 'non-existent-event-id';
+
+      mock.on(GetDistributionCommand).resolves(mockDistribution);
+      mock.on(UpdateDistributionCommand).resolves({});
+
+      await cloudFrontDistributionsManager.deleteOrigin({
+        eventId,
+      });
+
+      const getDistributionCall = mock
+        .calls()
+        .find((call) => call.args[0] instanceof GetDistributionCommand);
+      expect(getDistributionCall).toBeDefined();
+
+      const updateCall = mock
+        .calls()
+        .find((call) => call.args[0] instanceof UpdateDistributionCommand);
+      expect(updateCall).toBeDefined();
+
+      const updateInput = updateCall?.args[0]
+        .input as UpdateDistributionCommandInput;
+      expect(updateInput.DistributionConfig).toEqual(
+        mockDistribution.Distribution?.DistributionConfig
+      );
+    });
+  });
+});
+
+const setup = () => {
+  const client = new CloudFrontClient({
+    credentials: {
+      accessKeyId: 'fakeAccessKeyId',
+      secretAccessKey: 'fakeSecretAccessKey',
+    },
+  });
+  const cloudFrontDistributionsManager = new CloudFrontDistributionsManager({
+    client,
+    cdnDistributionId: process.env.DISTRIBUTION_ID || 'test-distribution-id',
```

Comment:
Can we remove all mentions of process.env is this file? It makes it confusing while it's not required

Reply:
You can set a distribution id in the function, and  return it, and use it in the tests
```typescript
return { client, cloudFrontDistributionsManager, distributionId };
```

```typescript
const { client, cloudFrontDistributionsManager, distributionId } = setup();
```